File: string/is_binary_number_multiple_3.cpp
//{ Driver Code Starts

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function template for C++
class Solution {
public:
  int isDivisible(string s) {
    // complete the function here
    int ans = 0;
    for (int i = s.size() - 1; i >= 0; --i)
      ans = (ans * 2 + s[i] - '0') % 3;
    return ans == 0;
  }
};

//{ Driver Code Starts.
int main() {
  int t;
  cin >> t;
  while (t--) {
    string s;
    cin >> s;
    Solution ob;
    cout << ob.isDivisible(s) << endl;
  }
  return 0;
}

// } Driver Code Ends

File: math/is_binary_number_multiple_3.py
# User function Template for python3
class Solution:
    def isDivisible(self, s):
        # code here
        return 0 if int(s, 2) % 3 else 1


# {
# Driver Code Starts

# Initial Template for Python 3

if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        s = input()
        ob = Solution()
        ans = ob.isDivisible(s)
        print(ans)

# } Driver Code Ends
